utils_cv/detection/model.py

```diff
@@ -100,18 +86,15 @@ def __init__(
         weight_decay: float = 0.0005,
     ):
         """ Initialize leaner object. """
-        self.device = (
-            torch.device("cuda")
-            if torch.cuda.is_available()
-            else torch.device("cpu")
-        )
+        self.device = torch_device()
         self.model = model
         self.dataset = dataset
         self.lr = lr
         self.momentum = momentum
         self.weight_decay = weight_decay
 
         # get training dataloaders and datasets
+        self.dataset = dataset
```

Comment:
I believe the line self.dataset=dataset is a duplicate. Also, should we remove setting train_ds, test_ds and train_dl since we can refer to them using the dataset object. That might also prevent confusion going forwards if somebody updates self.dataset but forgets to also update self.train_ds